book/modules/appendix3.tex

```diff
@@ -0,0 +1,172 @@
+Determinant is useful: it measures the volume distortion of a linear transformation, it helps us determine if a matrix is invertible of not, and it provides us with a simple formula for solving \(A\vec x=\vec y\) when \(\vec x\) is invertible.
+
+In Module 14, you will learn an algorithm for evaluating determinants via row reduction. However, for \(2\times 2\) and \(3\times 3\) determinants, we have methods which are more convenient than row reduction. We will explore these methods in this appendix.
+
+\Heading{A formula for \(2\times 2\) determinants}
+For a \(2\times 2\) matrix, once we know its four entries, we can calculate its determinant directly.
+\begin{theorem}[\(2\times 2\) determinant formula]
+  Let \(M\) be a \(2\times 2\) matrix. Suppose \(M=\mat{a & b \\ c & d}\). Then \[\det(M)=ad-bc.\]
+\end{theorem}
+\begin{example}
+Use the \(2\times 2\) determinant formula to find \(\det(M)\), where \(M=\mat{1 & 6 \\ 2  & 7}\).
+\end{example}
+\textbf{Solution. }Apply the formula, and we get \[\det(M)=1\times 7-2\times 6=-5.\]
+
+Notice that if we apply row reduction algorithm on \(M\), we get \[\mat{1 & -6 \\ 0 & 1}\mat{1 & 0 \\ 0 & -\frac{1}{5}}\mat{1 & 0 \\ -2 & 1}M=I_{2}.\]
+Thus, \[M=\mat{1 & 0 \\ 2 & 1}\mat{1 & 0 \\ 0 & -5}\mat{1 & 6 \\ 0 & 1}.\]
+Using theory developed in Module 14, this gives us that \(\det(M)=-5\). In other words, the result given by the formula agrees with the result given by row reduction.
+
+\Heading{Calculating \(3\times 3\) determinants (I): The diagonal trick}
+Now we turn to \(3\times 3\) determinants. We still have a manageable formula, although it is less appealing than the \(2\times 2\) formula:
+\begin{theorem}[\(3\times 3\) determinant formula]
+  Let \(M\) be a \(3\times 3\) matrix. Suppose \(M=\mat{a & b & c \\ d & e & f \\ g & h & i}\). Then \[\det(M)=aei+bfg+cdh-gec-hfa-idb.\]
+\end{theorem}
+You see, this formula is not easy to remember. To help you remember this formula, here's a trick called the \textit{diagonal trick}. Applying this formula, then, is no more than applying this trick on the matrix.
+
+Now we outline the procedure of calculating \(\det(\mat{a & b & c \\ d & e & f \\ g & h & i})\) via this trick.
+
+\textbf{Step 1. }Copy the matrix, and then copy the \emph{first two columns} to the right of the matrix:
+\[\begin{NiceArray}{CCCCCCCCCCC}[margin]
```

Comment:
Why don't you do this with the augmented matrix syntax? Then you can use `{\color{...} ...}` in the cells to distinguish the columns/diagonals.